tests/test_distributed.py

```diff
@@ -183,8 +189,14 @@ def test_no_model_parallel(self):
             'transformer/classifier',
         ]:
             config['model'] = m
-            with self.assertRaises(RuntimeError):
```

Comment:
Is there a reason for replacing this? Is throwing the exception here indeterministic? Because what you do next is checking for this error again.

Reply:
It's just to get the `finally` clause which is necessary tearDown. I can't use tearDown itself because the retry decorator doesn't call it.